lib/src/specs/class.dart

```diff
@@ -60,6 +69,23 @@ abstract class Class extends Object
       visitor.visitClass(this, context);
 }
 
+enum ClassModifier {
+  base,
+  final$,
+  interface;
+
+  String get name {
```

Comment:
Dunno if this is Dart 3.0 code yet, but you can use a fancy switch expression as well:

```dart
String get name =>
    return switch (this) {
      base => 'base',
      final$ => 'final',
      interface => 'interface',
    };

Reply:
Good catch! I did try bumping the constraints to ^3.0.0 but it will require a larger effort as the repo depends on some of Dart 2 behavior like mixing in abstract classes.